aiutils/tftools/batch_normalizer.py

```diff
@@ -0,0 +1,94 @@
+"""A helper class for managing batch normalization state.
+
+This class is designed to simplify adding batch normalization
+(http://arxiv.org/pdf/1502.03167v3.pdf) to your model by
+managing the state variables associated with it.
+
+Important use note:  The function get_assigner() returns
+an op that must be executed to save the updated state.
+A suggested way to do this is to make execution of the
+model optimizer force it, e.g., by:
+
+  update_assignments = tf.group(bn1.get_assigner(),
+                                bn2.get_assigner())
+  with tf.control_dependencies([optimizer]):
+    optimizer = tf.group(update_assignments)
+"""
+
+import tensorflow as tf
+from tensorflow.python import control_flow_ops
+
+class BatchNormalizer(object):
+    """Helper class that groups the normalization logic and variables.
+
+    Use:
+        ewma = tf.train.ExponentialMovingAverage(decay=0.99)
+        bn = BatchNormalizer(input, 0.001, ewma, True)
+        update_assignments = bn.get_assigner()
+        x = bn.normalize(y, train=training?)
+        (the output x will be batch-normalized).
+    """
+
+    def __init__(self, input, epsilon, ewma_trainer, name):
+        rank = len(input.get_shape().as_list())
+        in_dim = input.get_shape().as_list()[-1]
+
+        if rank == 2:
+            self.axes = [0]
+        elif rank == 4:
+            self.axes = [0, 1, 2]
+        else:
+            raise ValueError('Input tensor must have rank 2 or 4.')
+
+        self.mean = tf.Variable(tf.constant(0.0, shape=[in_dim]),
+                trainable=False)
+        self.variance = tf.Variable(tf.constant(1.0, shape=[in_dim]),
+                trainable=False)
+        with tf.variable_scope(name):
+            self.beta = tf.get_variable('offset',
```

Comment:
Can we call these `offset` and `scale` just so I don't have to remember which is beta and gamma?